ledger/ledger-on-memory/src/main/scala/com/daml/ledger/on/memory/InMemoryBatchedLedgerReaderWriter.scala

```diff
@@ -126,24 +118,56 @@ object InMemoryBatchedLedgerReaderWriter {
           metrics,
           inStaticTimeMode = needStaticTimeModeFor(timeProvider))
       val validator = BatchedSubmissionValidator[Index](
-        BatchedSubmissionValidatorParameters.default,
+        BatchedSubmissionValidatorFactory.defaultParametersFor(
+          batchingLedgerWriterConfig.enableBatching),
         keyValueCommitting,
         new ConflictDetection(metrics),
         metrics,
-        engine)
-      Resource.successful(
+        engine
+      )
+      val committer =
+        BatchedValidatingCommitter[Index](
+          () => timeProvider.getCurrentTime,
+          validator,
+          stateValueCache)
+      val readerWriter =
         new InMemoryBatchedLedgerReaderWriter(
           participantId,
           ledgerId,
-          () => timeProvider.getCurrentTime,
           dispatcher,
           state,
-          validator,
+          committer,
           metrics
-        ))
+        )
+      if (batchingLedgerWriterConfig.enableBatching) {
```

Comment:
I'm confused. We already use the `BatchedValidatingCommitter` here. Does this not do batching unless we also use `BatchedSubmissionValidatorFactory`?

Reply:
You need both in order to have the performance benefits. I.e., batched submission validation requires two parts: 1) creating batched submissions at the ledger API server side and 2) validating individual submissions within a batched submission in parallel in the validator/committer. 1) is ensured by `BatchingLedgerWriter` and 2) is enabled by `BatchedValidatingCommitter`. `ledger-on-memory` is a ledger which has the submitter and the committer in-process, hence, the above separation may be confusing (but is required for supporting production ready ledgers).

Reply:
Sounds good. Could you add a comment along these lines please?

Reply:
Added.